jobs/bury_treasure_addresses.go

```diff
@@ -554,3 +514,49 @@ func waitForConfirmation(treasureToBury models.Treasure, txHash string, txNonce
 		oyster_utils.LogIfError(err, nil)
 	}
 }
+
+func updateStatusSuccess(txType services.TxType, treasureRow models.Treasure) models.PRLStatus {
+	var newStatus models.PRLStatus
+	switch txType {
+	case services.PRLTransfer:
+		if treasureRow.PRLStatus == models.PRLPending || treasureRow.PRLStatus == models.PRLError {
+			newStatus = models.PRLConfirmed
+		}
+	case services.EthTransfer:
+		if treasureRow.PRLStatus == models.GasPending || treasureRow.PRLStatus == models.GasError {
+			newStatus = models.GasConfirmed
+		}
+	case services.PRLBury:
+		if treasureRow.PRLStatus == models.BuryPending || treasureRow.PRLStatus == models.BuryError {
+			newStatus = models.BuryConfirmed
+		}
+	default:
+		logInvalidTxType()
+	}
+	return newStatus
+}
+
+func updateStatusFailed(txType services.TxType, treasureRow models.Treasure) models.PRLStatus {
+	var newStatus models.PRLStatus
+	switch txType {
+	case services.PRLTransfer:
+		if treasureRow.PRLStatus == models.PRLPending {
+			newStatus = models.PRLError
+		}
+	case services.EthTransfer:
+		if treasureRow.PRLStatus == models.GasPending {
+			newStatus = models.GasError
+		}
+	case services.PRLBury:
+		if treasureRow.PRLStatus == models.BuryPending {
+			newStatus = models.BuryError
+		}
+	default:
+		logInvalidTxType()
+	}
+	return newStatus
+}
+
+func logInvalidTxType() {
+	oyster_utils.LogIfError(errors.New("not a valid tx type in bury_treasure_addresses waitForConfirmation"), nil)
```

Comment:
you probably want to include the txType/prlStatus value for debugging purpose.

Like:
oyster_utils.LogIfError(/*error*/, map[string]interface{}{"TxType" : txType, "PrlStatus" : treasureRow.PRLStatus}